### PR TITLE
ISPN-14256 CLI should work with piped input

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-14256

### CODE REVIEW

File: cli/src/main/java/org/infinispan/cli/util/ThrowingSupplier.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.infinispan.cli.util;
+
+/**
+ * Supplier that can throw any exception.
+ *
+ * @author Tristan Tarrant
+ * @since 10.0
+ */
+@FunctionalInterface
+public interface ThrowingSupplier<T> {
```
Comment (wburns) [inline cli/src/main/java/org/infinispan/cli/util/ThrowingSupplier.java:10]: This is just Callable btw

**Review by wburns — APPROVED** [review]

File: cli/src/main/java/org/infinispan/cli/util/ThrowingSupplier.java (hunk 1 above)
Comment (tristantarrant) [reply cli/src/main/java/org/infinispan/cli/util/ThrowingSupplier.java]: Right, removed

**Comment by wburns** [comment]
Integrated into main, thanks @tristantarrant !

**Comment by wburns** [comment]
@tristantarrant  Looks like you wanted to add a 14.0.x branch PR?

**Pull request merged**